patches/tModLoader/Terraria/ModLoader/UI/Interface.cs

```diff
@@ -426,29 +427,27 @@ internal static void ServerModMenu(out bool reloadMods)
 				exit = true;
 			}
 			else if (command.StartsWith("c")) {
-				int modIndex = Convert.ToInt32(command[2..]) - 1;
-				if (modIndex < mods.Length) {
-					if (ModLoader.TryGetMod(mods[modIndex].Name, out Mod mod)) {
-						if (ConfigManager.Configs.TryGetValue(mod, out List<ModConfig> configs) && configs.Any(config => config.Mode == ConfigScope.ServerSide)) {
-							ConfigureMod(mod, configs);
+				Match match = new Regex("c (\\d+)").Match(command);
+				if (match.Success) {
+					int modIndex = Convert.ToInt32(match.Groups[1].Value) - 1;
+					if (modIndex >= 0 && modIndex < mods.Length) {
+						if (ModLoader.TryGetMod(mods[modIndex].Name, out Mod mod)) {
+							if (ConfigManager.Configs.TryGetValue(mod, out List<ModConfig> configs) && configs.Any(config => config.Mode == ConfigScope.ServerSide)) {
+								// We are acting on the actual configs rather than a clone because a reload will be forced anyway. If changing configs during server play is implemented later this will need to adjust to the clone approach.  
+								ConfigureMod(mod, configs);
+							}
+							else {
+								WriteColoredLine(ConsoleColor.Yellow, Language.GetTextValue("tModLoader.DedErrorNoConfig"));
+							}
 						}
 						else {
-							Console.ForegroundColor = ConsoleColor.Yellow;
-							Console.WriteLine(Language.GetTextValue("tModLoader.DedErrorNoConfig"));
-							Console.ResetColor();
+							WriteColoredLine(ConsoleColor.Yellow, Language.GetTextValue("tModLoader.DedErrorNotEnabled"));
 						}
 					}
 					else {
-						Console.ForegroundColor = ConsoleColor.Yellow;
-						Console.WriteLine(Language.GetTextValue("tModLoader.DedErrorNotEnabled"));
-						Console.ResetColor();
+						WriteColoredLine(ConsoleColor.Yellow, Language.GetTextValue("tModLoader.DedErrorModOOB"));
 					}
 				}
```

Comment:
This style of code needs to be flattened. Error cases should come first, and fail fast. You may be able to achieve this with an `else if` chain and pattern matching. (Eg `Convert.ToInt32(match.Groups[1].Value) - 1 is int modIndex && ...`), otherwise, extract it to a separate method and use early `return`